core/variant.cpp

```diff
@@ -1575,7 +1575,14 @@ Variant::operator String() const {
 				};
 #endif
 				if (_get_obj().obj->has_method("_to_string")) {
-					return *reinterpret_cast<const String *>(_get_obj().obj->call("_to_string")._data._mem);
+					Variant::CallError ce;
+					Variant ret = _get_obj().obj->call("_to_string", NULL, 0, ce);
+
+					// Make sure a Variant::String was returned
+					if (ce.error == Variant::CallError::CALL_OK && ret.get_type() == Variant::STRING) {
+						return *reinterpret_cast<const String *> (ret._data._mem);
```

Comment:
You dont need to do this, just return it and that's it.. or do operator String().